esp-hal/src/uart.rs

```diff
@@ -544,6 +550,13 @@ where
             }
         }
 
+        // Setting err_wr_mask stops uart from storing data when data is wrong according
+        // to reference manual
+        T::register_block()
+            .conf0()
+            .modify(|_, w| w.err_wr_mask().set_bit());
+        // TODO reset? https://github.com/espressif/esp-idf/blob/4d90eedb6ef32ff949c6f298a3d845f9f07bac92/components/hal/esp32s3/include/hal/uart_ll.h#L112
```

Comment:
IIRC that is done in `uart_peripheral_reset` which is called by `init` earlier. There were problems with UART0 if console logging is enabled

Reply:
Good spot, thank you! I've removed the comment.